### ISSUE TITLE
Turn off "Compiling additional model methods..." message when calling `init_model_methods`

### ISSUE THREAD

**Issue by ben18785** [stan-dev/cmdstanr#914]
**Is your feature request related to a problem? Please describe.**
I am creating a package that repeatedly calls `init_model_methods` (each time I use a different dataset with my Stan model) to give access to `log_prob`. Each time it does so, it outputs

> Compiling additional model methods...

to the console.

**Describe the solution you'd like**
I'd suggest either removing this informative message or allowing users to do so.

**Describe alternatives you've considered**
I could capture the output in my package and then remove it from users' view, but that feels messy.

**Comment by jgabry** [comment]
Maybe this can be handled as part of https://github.com/stan-dev/cmdstanr/pull/894

**Comment by avehtari** [comment]
It seems #894 has been superseded by #1083 which has not yet been merged

**Comment by jgabry** [comment]
Yes, there are still some failing tests and a few merge conflicts. I think @andrjohns has been very busy, but maybe he will have time to finish it at some point. Or maybe someone else can finish it.

**Comment by jgabry** [comment]
Finally dealing with this in #1285 since there hasn't been progress on those other PRs. Sorry for the long delay!!
